### PR TITLE
refactor: Field selection api for search

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -351,12 +351,14 @@ message SearchRequest {
   bytes facet = 6;
   // Array of fields and corresponding sort orders to order the results `[{ "salary": "$desc" }]`
   bytes sort = 7;
-  // Fetch specific fields from a document. Default is all
-  bytes fields = 8;
+  // Array of document field names to include in results. By default, all documents are included.
+  repeated string include = 8;
```
Comment (adilansari) [inline server/v1/api.proto:355]: Should we go `include` or `include_fields`?

Comment (JigarJoshi) [reply server/v1/api.proto:355]: I prefer later.

File: server/v1/api.proto [hunk 2]
```diff
@@ -351,12 +351,14 @@ message SearchRequest {
...
+  // Array of document field names to exclude from results. `Include`, if specified, takes precedence over exclude.
```
Comment (JigarJoshi) [inline server/v1/api.proto:356]: nit: `takes precedence of include`

**Review by JigarJoshi — APPROVED** [review]

File: server/v1/api.proto [hunk 3]
```diff
@@ -351,12 +351,14 @@ message SearchRequest {
   bytes facet = 6;
   // Array of fields and corresponding sort orders to order the results `[{ "salary": "$desc" }]`
   bytes sort = 7;
-  // Fetch specific fields from a document. Default is all
-  bytes fields = 8;
+  // Array of document field names to include in results. By default, all documents are included.
```
Comment (JigarJoshi) [inline server/v1/api.proto:354]: nit: `By default, all fields are included.`

**Review by himank — APPROVED** [review]

**Pull request merged**